src/spikeinterface/qualitymetrics/pca_metrics.py

```diff
@@ -156,16 +159,16 @@ def compute_pc_metrics(
         )
         items.append(func_args)
 
-    if not run_in_parallel:
+    if not run_in_parallel and metric_names:
```

Comment:
what does this do? I guess a user could supply metrics_names = [] as an empty list, but if it is None it is supplied. Just a little confused about this.

Reply:
The nearest neighbour metrics are stripped out of metric_names earlier (line 114). [Ultimately this is because the non-nn metrics are parallelised using`pca_metrics_one_unit` while the nn ones must be calculated in a different way.]

So this is here to avoid running the code underneath in the case of a user just running nn metrics. Without it, if you just run nn metrics the user still sees a progress bar saying "calculating pca metrics...". 

Might be clearer to make a new list called `non_nn_metric_names`?

Reply:
Yeah I think that would be clearer from a developer perspective. But I guess if it works it works.

Reply:
I think it is clearer: have done it